private.cpp

```diff
@@ -218,18 +228,33 @@ void _DroidMediaBufferQueue::frameAvailable() {
   }
 }
 
-void _DroidMediaBufferQueue::buffersReleased() {
-  for (int i = 0; i < android::BufferQueue::NUM_BUFFER_SLOTS; ++i) {
-    DroidMediaBufferSlot &slot = m_slots[i];
-    slot.droidBuffer.clear();
-    slot.mGraphicBuffer = 0;
+void _DroidMediaBufferQueue::releaseBufferResources(DroidMediaBuffer *buffer) {
+  if (m_unslotted.find(buffer) != m_unslotted.end()) {
+    ALOGI("Unslotted buffer, resources already released %" PRIxPTR, (uintptr_t)buffer);
+  } else {
+    buffer->decStrong(0);
+    buffer->m_buffer.clear();
   }
+}
 
+void _DroidMediaBufferQueue::buffersReleased() {
   android::AutoMutex locker(&m_lock);
 
   if (m_data) {
     m_cb.buffers_released(m_data);
   }
+
+  // Releasing DroidMediaBuffer memory for slots
+  for (int i = 0; i < android::BufferQueue::NUM_BUFFER_SLOTS; ++i) {
+      DroidMediaBufferSlot &slot = m_slots[i];
+      slot.droidBuffer.clear();
+      slot.mGraphicBuffer = 0;
```

Comment:
Nitpick: indentation in this for loop does not match the surrounding code (4 spaces vs 2 spaces).